wrangler/src/txFunctions/runCB.js

```diff
@@ -0,0 +1,197 @@
+  const { txFunctionHash } = params
+
+  const { value, metadata } = await TX_FUNCTIONS.getWithMetadata(txFunctionHash, 'arrayBuffer')
+
+  if (!value)
+    throw {status: 404, message: `txFunction could not be found this turret`}
+
+  const { length, txFunctionSignerPublicKey, txFunctionSignerSecret } = metadata
+
+  const txFunctionBuffer = Buffer.from(value)
+  const txFunction = txFunctionBuffer.slice(0, length).toString()
+
+  const body = await request.json()
+  const claimableBalanceToken = request.headers.get('authorization')?.split(' ')?.[1]
+
+  if (!claimableBalanceToken)
+    throw {message: `txFunctionFee is missing`}
+
+  const [
+    claimableBalanceId, 
+    claimableBalanceSignature, 
+    ...txFunctionHashes
+  ] = JSON.parse(Buffer.from(claimableBalanceToken, 'base64'))
+
+  if (
+    txFunctionHashes.length
+    && txFunctionHashes.indexOf(txFunctionHash) === -1
+  ) throw {message: `txFunctionFee is invalid for this txFunction`}
+
+  const { metadata: feeMetadata } = await TX_FEES.getWithMetadata(claimableBalanceId)
+
+  let feeTotalBigNumber
+  let feeSpentBigNumber
+
+  if (feeMetadata) {
+    feeTotalBigNumber = new BigNumber(feeMetadata.total)
+    feeSpentBigNumber = new BigNumber(feeMetadata.spent)
+
+    if (feeSpentBigNumber.isGreaterThanOrEqualTo(feeTotalBigNumber)) {
+      event.waitUntil(txSponsorsSettle(claimableBalanceId))
+      throw {status: 402, message: `txFunctionFee has been spent`}
+    }
+  }
+
+  else {
+    const { asset, amount, sponsor, claimants } = await fetch(`${HORIZON_URL}/claimable_balances/${claimableBalanceId}`)
+    .then(async (res) => {
+      if (res.ok)
+        return res.json()
+      throw res
+    })
+
+    const sponsorKeypair = Keypair.fromPublicKey(sponsor)
+
+    if (!(
+      asset === 'native'
+      && new BigNumber(amount).isGreaterThanOrEqualTo(XLM_FEE_MIN)
+      && new BigNumber(amount).isLessThanOrEqualTo(XLM_FEE_MAX)
+      && sponsorKeypair.verify(JSON.stringify([
+        claimableBalanceId,
+        ...txFunctionHashes
+      ]), Buffer.from(claimableBalanceSignature, 'hex'))
+      && claimants.length <= 2
+      && loFind(claimants, (claimant) => 
+        claimant.destination === TURRET_ADDRESS
+        && claimant.predicate.unconditional
+      )
+      && (
+        claimants.length === 2
+        ? loFind(claimants, (claimant) => 
+          claimant.destination === sponsor
+          && claimant.predicate?.not?.abs_before
+          && moment.utc(claimant.predicate.not.abs_before).subtract(180, 'days').isAfter() // TODO: Days till claimable balance can be recovered by sponsor. Should probably be a configurable TURRET setting
+        ) : true
+      )
+      && claimants.length <= 2
+      && claimants[0]?.destination === TURRET_ADDRESS
+      && claimants[0]?.predicate?.unconditional
+    )) throw {message: `txFunctionFee is invalid`}
+
+    feeTotalBigNumber = new BigNumber(amount)
+    feeSpentBigNumber = new BigNumber(0)
```

Comment:
is there a reason we are overwriting the `feeSpentBigNumber` here?

Reply:
It hasn't been set yet due to the if / else block so it's not really overwriting, it's just setting it. This is the else block if the claimable balances is brand new and hasn't been used yet.